### ISSUE TITLE
Support language <X>

### ISSUE THREAD

**Issue by sgraf812** [csmith-project/creduce#153]
I'd like to use C-reduce to reduce Haskell programs. What would need to be changed to completely disable any language-specific passes and only include passes specific to Haskell?

I figured that instead of rewriting everything from scratch, I'd better benefit from the many man hours that already went into C-reduce.

**Comment by regehr** [comment]
I'll patch C-Reduce to make this happen. In the meantime, however, there is no problem leaving those passes enabled. They won't do anything useful but they won't hurt either, other than wasting a bit of CPU time.

**Comment by regehr** [comment]
Also, the job of reducing a strongly typed language is harder than reducing something like C. It may be the case that you end up wanting to write Haskell-specific passes that perform desirable transformations. If you do this and want to integrate them into C-Reduce, let's talk.

**Comment by sgraf812** [comment]
> Haskell-specific passes

Yeah, that's probably what needs to be done to be useful. Let's see what I can whip up.

**Comment by regehr** [comment]
Please try out C-Reduce as-is and see if it does anything useful. It has been known to do good reduction of Rust code, for example. Also if you have any GHC crashers (or whatever) we'd be interested to see them, if they can be reproduced relatively easily.

**Comment by regehr** [comment]
OK, if you grab the master branch there's a --not-c command line flag that turns off some of the passes that are the most specific to C/C++.
